bin/varnishncsa/varnishncsa.c

```diff
@@ -166,15 +166,39 @@ static struct ctx {
 
 enum format_policy {
         FMTPOL_FIRST = 0,
+        FMTPOL_REQ,
+        FMTPOL_RESP,
+        FMTPOL_BEREQ,
+        FMTPOL_BERESP,
         FMTPOL_ALWAYS
 };
 
+#define FMTPOL_REQ_SIDE() *CTX.side == 'c' ? FMTPOL_REQ : FMTPOL_BEREQ
+#define FMTPOL_RESP_SIDE() *CTX.side == 'c' ? FMTPOL_RESP : FMTPOL_BERESP
```

Comment:
The enum looks a lot like what I suggested in a rushed review, so I'd like to apologize upfront for the alternative I'm going to suggest.

In the proposal from https://github.com/varnishcache/varnish-cache/issues/3528#issuecomment-2017621182 4 categories emerged:

- what we receive
- what we send
- internal events
- other

We should probably map the enum to the actual policies:

```c
enum format_policy {
        FMTPOL_SENT,
        FMTPOL_RECD,
        FMTPOL_INTERNAL,
};
```

Composing `CTX.side` with the format policy should be enough to circle back to which message was sent or received in `process_format_policy()`.